meson.build

```diff
@@ -0,0 +1,172 @@
+
+bison = find_program('bison', version : '>3.7')
+
+reflex_dep = dependency('reflex')
+reflex = find_program('reflex')
+
+cc = meson.get_compiler('cpp')
+
+# Silence warnings from reflex headers
+add_project_arguments(cc.get_supported_arguments('-Wno-non-virtual-dtor'), language : 'cpp')
+
+deps = [
+  absl_container_dep,
+  absl_hash_dep,
+  absl_status_dep,
+  absl_strings_dep,
+  absl_time_dep,
+  mpdecimal_dep,
+  protobuf_dep,
+  re2_dep,
+  reflex_dep,
+]
+
+includes = include_directories('.')
+
+proto = files(
+  'beancount/ccore/data.proto',
+  'beancount/ccore/date.proto',
+  'beancount/ccore/number.proto',
+  'beancount/cparser/ledger.proto',
+  'beancount/cparser/options.proto',
+  'beancount/cparser/parser.proto',
+)
+
+protoc_gen = generator(protoc,
+  output : ['@BASENAME@.pb.cc', '@BASENAME@.pb.h'],
+  arguments : ['--proto_path=@CURRENT_SOURCE_DIR@', '--cpp_out=@BUILD_DIR@', '@INPUT@'])
+
+proto_cc = protoc_gen.process(proto,
+  preserve_path_from : meson.current_source_dir())
+
+subdir('beancount/cparser')
+
+# reflex_gen = generator(reflex,
+#   output : ['@BASENAME@.cc', '@BASENAME@.h'],
+#   arguments : ['--unicode', '--full', '--header-file=@OUTPUT1@', '-o', '@OUTPUT0@', '@INPUT@'])
+
+# scanner_cc = reflex_gen.process('beancount/cparser/scanner.lxx',
+#   preserve_path_from : meson.current_source_dir())
+
+# bison_gen = generator(bison,
+#   output : ['@BASENAME@.cc', '@BASENAME@.h'],
+#   arguments : ['--report=itemset', '--verbose', '-Wall', '-Werror',
+#                '--defines=@OUTPUT1@',
+#                '--define=api.location.file="location.h"',
+#                '-o', '@OUTPUT0@', '@INPUT@'])
+
+# parser_cc = bison_gen.process('beancount/cparser/parser.yxx',
+#   preserve_path_from : meson.current_source_dir())
+
+sources = files(
+  'beancount/ccore/account.cc',
+  'beancount/ccore/account_types.cc',
+  'beancount/ccore/data.cc',
+  'beancount/ccore/date.cc',
+  'beancount/ccore/inventory.cc',
+  'beancount/ccore/number.cc',
+  'beancount/ccore/std_utils.cc',
+  'beancount/cparser/builder.cc',
+  'beancount/cparser/ledger.cc',
+  'beancount/cparser/test_utils.cc',
+)
+
+cbeancount = static_library('cbeancount',
+  sources, proto_cc, scanner_cc, parser_cc,
+  include_directories : includes,
+  dependencies : deps)
+
+tests = [
+  'beancount/ccore/account_test.cc',
+  'beancount/ccore/date_test.cc',
+  'beancount/ccore/number_test.cc',
+  'beancount/ccore/std_utils_test.cc',
+  'beancount/cparser/parser_test.cc',
+  'beancount/cparser/scanner_test.cc',
+  'beancount/cparser/test_utils_test.cc',
+]
+
+foreach test : tests
+  t = executable(
+    test.underscorify(),
+    test,
+    include_directories : [
+      includes,
+      cbeancount.private_dir_include()
+    ],
+    link_with : cbeancount,
+    dependencies : [
+      deps,
+      gmock_dep,
+      gtest_main_dep,
+    ])
+
+  test(test.split('.')[0], t, protocol : 'gtest')
+endforeach
+
+python.extension_module(
+  'beancount.ccore._core',
```

Comment:
This seems wrong, you should be installing a `_core` extension_module into the `beancount.ccore` install_dir.

Python doesn't expect `site-packages/beancount.ccore._core.{SOEXT}`, but rather, expects `site-packages/beancount/ccore/_core.{SOEXT}`

Reply:
Indeed. Thank you @eli-schwartz for having a look. When I was working on this I used a local hack that uses meson as a setuptool backend for compilation to install or produce wheels and it handled placing the extension in the right place. I cannot remember now if naming the extension like that was required to make the hack work. I definitely should revisit that and publish it somewhere.

By the way, do you have an interest on Beancount or are you reviewing all PRs on Github that mention Meson? 😃

Reply:
I'm (trying to) review all PRs on GitHub that mention Meson. 😀